packages/pouchdb-core/src/changes.js

```diff
@@ -58,7 +59,11 @@ function Changes(db, opts, callback) {
     if (opts.isCancelled) {
       return;
     }
-    self.emit('change', change);
+    try {
+      self.emit('change', change);
+    } catch (e) {
+      guardedConsole('error', 'Error in .on("change", function):', e);
+    }
```

Comment:
try/catches should be isolated to avoid V8 deopts, but I can fix this on the way in